### DISCUSSION TITLE
woks in AMD 6600 + Rocm 5.7.1

### DISCUSSION THREAD

**Discussion by Consti-Nic** [neonbjb/tortoise-tts#739] [Show and tell]
Hi, i managed to get TorToise working on my 6600 (no xt, sadly) in Debian 12 using Rocm 5.7.1 
following both this discussion and this issue:
https://github.com/neonbjb/tortoise-tts/discussions/670
https://github.com/neonbjb/tortoise-tts/issues/366

### Important: 
Because i use Debian 12, first thing i did was to change global python version to 3.10.9 with pyenv as python 3.11 is not supported by Rocm, then added amdgpu-install repo to install rocm:
https://blog.moubou.com/2023/11/install-amd-rocm-on-debian-bookworm/

i was too lazy to use docker, so venv instead.

### Also important:
for some reason my pip kept pulling "torchaudio-2.2.1-cp310-cp310-manylinux1_x86_64.whl" and "torch-2.2.1-cp310-cp310-manylinux1_x86_64.whl". With these wheels tortoise doesn't recognize the gpu , uses cpu instead and  throws  
```
torch/amp/autocast_mode.py:204: UserWarning: User provided device_type of 'cuda', but CUDA is not available. Disabling
  warnings.warn('User provided device_type of \'cuda\', but CUDA is not available. Disabling')
```
i manually downloaded torch and torch audio from https://download.pytorch.org/whl/rocm5.2 to my tortoise folder and added them to requirements.txt

download these two:

torch-2.2.0+rocm5.7-cp310-cp310-linux_x86_64.whl
torchaudio-2.2.0+rocm5.7-cp310-cp310-linux_x86_64.whl

I'll probably upload these files to a github repo and link the release in my requirements for pip to use with docker in the future

newer versions of rotary_embedding_torch threw incompatible version error with einops 0.4.1, downgraded versions until 0.2.6 which seemed to work 

Requirements.txt: 
```
tqdm
rotary_embedding_torch==0.2.6
transformers==4.31.0
tokenizers==0.13.3
inflect
progressbar
einops==0.4.1
unidecode
scipy
librosa==0.9.1
ffmpeg
numpy
numba
appdirs
--extra-index-url https://download.pytorch.org/whl/rocm5.2
/media/user/discoD/tts/tortoise-tts/torch-2.2.0+rocm5.7-cp310-cp310-linux_x86_64.whl
/media/user/discoD/tts/tortoise-tts/torchaudio-2.2.0+rocm5.7-cp310-cp310-linux_x86_64.whl
threadpoolctl
llvmlite
appdirs
nbconvert==5.3.1
tornado==4.2
pydantic==1.9.1
deepspeed==0.8.3
py-cpuinfo
hjson
psutil
sounddevice
```

Capiton:
![Screenshot_2024-02-26_11-29-14](https://github.com/neonbjb/tortoise-tts/assets/125901616/787672d8-2d87-435a-bd05-a42516979ace)

PC stuff:
![Screenshot_2024-02-26_11-43-54](https://github.com/neonbjb/tortoise-tts/assets/125901616/354062c6-b4d4-4a7e-a1d6-75f971108b14)

added on top of  **do_tts** to quickly verify if it uses gpu
```
 if torch.cuda.is_available():
    print("CUDA GPU is available!")
else:
    print("CUDA GPU is not available.")
```

**Comment by AVIER0** [comment]
any idea how to run it on windows
